### ISSUE TITLE
Exception when recreating file opened in klogg

### ISSUE THREAD

**Issue by kindacute** [variar/klogg#451]
Details for the issue
--------------------

I just recreated a file that was opened in klogg

-------------------------
Exception:
PartialIndexOperation failed: vector
-------------------------
Useful extra information
-------------------------
> Klogg version 22.02.0.1199 (built on 2022-02-20 from commit 4e40303) [built for x86_64-little_endian-lp64]
> running on macOS 12.2 (darwin/21.3.0) [x86_64]
> and Qt 5.15.2

**Comment by variar** [comment]
Looks like this can be reproduced on Windows (#455). I'm investigating the root cause

**Comment by variar** [comment]
Unfortunately, I can't reproduce the issue. However, I think I've found the possible cause. @kindacute @JaimeNA03 could you try build 22.03.0.1210?

**Comment by variar** [comment]
As mentioned in #466 this can happen when polling is enabled

**Comment by variar** [comment]
Unfortunately, I can't reproduce it. It the issue is still there please enable klogg logs and attach them here.

**Comment by kindacute** [comment]
@variar please describe how to enable logging?

**Comment by variar** [comment]
@kindacute it should be in File->Preferences->Advanced tab. Verbosity 4 should be enough. Log file is created in tmp folder with name starging with `klogg_`. 

Please update to the latest CI build, the exception message there could be more precise.

**Comment by kindacute** [comment]
Unfortunately, I cannot update to the latest CI build (from git repo, right?). Looks like readme is outdated. QT version mismatch is a culprit.

**Comment by variar** [comment]
> Unfortunately, I cannot update to the latest CI build (from git repo, right?). Looks like readme is outdated. QT version mismatch is a culprit.

@kindacute You mean building from source doesn't work or installing from dmg https://github.com/variar/klogg/releases/tag/continuous-osx is broken?

**Comment by variar** [comment]
> Ok, here is the log: https://drive.google.com/file/d/1UQh3EtUt6RJbWWWNl3xg9wq_hpcwUi51/view?usp=sharing

Thanks! This confirms my initial guess about the place in code where this error occurs:
```
2022-06-06T18:18:07 critical [170768] [CompressedLinePositionStorage::at@292] Line number not in storage: 18446744073709551615, storage size is 0
2022-06-06T18:18:07 info [370785] [IndexOperation::readFileInBlocks@520] IO thread done
2022-06-06T18:18:07 critical [170768] [PartialIndexOperation::run@775] "PartialIndexOperation failed: Line number not in storage"
```

Now I need to figure out why it is happening.

**Comment by variar** [comment]
> Ok, here is the log: https://drive.google.com/file/d/1UQh3EtUt6RJbWWWNl3xg9wq_hpcwUi51/view?usp=sharing

@kindacute I've downloaded the log file. Link can be removed.

**Comment by variar** [comment]
The issue was happening in a specific case. Klogg has to open a file that has exactly one line and that line doesn't contain line-feed character. If some external program writes to this file, klogg tries to reindex it but fails with off-by-one error (unsinged line counter wraps around). I've added a unit test for this case, and a fix should be in 22.06.0.1255.

@kindacute  thanks for you help!

**Comment by variar** [comment]
As for build from source issue. Looks like Qt was build for mac os target 10.15, and by default klogg is buiding for 10.14. I've mentioned this in build docs. And changed default behaviour to target 10.15 if klogg is built with Qt6.
